freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_install.sh

```diff
@@ -23,4 +23,13 @@ ipa-server-install \
           --auto-forwarders \
           --unattended
 
+# Setup basic CNAME records for pointing to FreeIPA services
+echo "$FPW" | kinit admin
+ipa dnsrecord-add $DOMAIN kdc --cname-rec="ipa-ca.$DOMAIN."
+ipa dnsrecord-add $DOMAIN kerberos --cname-rec="ipa-ca.$DOMAIN."
+ipa dnsrecord-add $DOMAIN ldap --cname-rec="ipa-ca.$DOMAIN."
+ipa dnsrecord-add $DOMAIN freeipa --cname-rec="ipa-ca.$DOMAIN."
+ipa dnsrecord-add $DOMAIN ipa --cname-rec="ipa-ca.$DOMAIN."
+kdestroy
```

Comment:
please do it from java

Reply:
I would like to understand why we needed to take something that was 8 lines in a salt script, was only going to be run once on an install of the server and move it to Java.  Now we have a lot more code to maintain.  Isn't this what using salt as the node management supposed to give us.  I am just wanting to understand where the demarkation oh what is supposed to be controlled by java and what the salt scripts are supposed to do.

Reply:
Right now we cannot update the Salt states on an existing cluster. If we need to change here something that we cannot do.

Reply:
Think of it as Cloudera Manager. We could install a cluster from Salt also, but we use its API. We would lose lots of capabilities if we would implement everything in Salt.